datadog-integrations-aws-handler/src/datadog_integrations_aws/handlers.py

```diff
@@ -0,0 +1,191 @@
+LOG = logging.getLogger(__name__)
+TYPE_NAME = "Datadog::Integrations::AWS"
+TELEMETRY_TYPE_NAME = "integrations-aws"
+
+resource = Resource(TYPE_NAME, ResourceModel)
+test_entrypoint = resource.test_entrypoint
+
+
+def build_aws_account_from_model(model):
+    aws_account = AWSAccount()
+    if model.AccountID is not None:
+        aws_account.account_id = model.AccountID
+    if model.RoleName is not None:
+        aws_account.role_name = model.RoleName
+    if model.AccessKeyID is not None:
+        aws_account.access_key_id = model.AccessKeyID
+    if model.HostTags is not None:
+        aws_account.host_tags = model.HostTags
+    if model.FilterTags is not None:
+        aws_account.filter_tags = model.FilterTags
+    if model.AccountSpecificNamespaceRules is not None:
+        aws_account.account_specific_namespace_rules = model.AccountSpecificNamespaceRules
+    return aws_account
+
+
+@resource.handler(Action.CREATE)
+def create_handler(
+        session: Optional[SessionProxy],
+        request: ResourceHandlerRequest,
+        callback_context: MutableMapping[str, Any],
+) -> ProgressEvent:
+    LOG.info("Starting %s Create Handler", TYPE_NAME)
+    model = request.desiredResourceState
+
+    aws_account = build_aws_account_from_model(model)
+    with v1_client(
+            model.DatadogCredentials.ApiKey,
+            model.DatadogCredentials.ApplicationKey,
+            model.DatadogCredentials.ApiURL or "https://api.datadoghq.com",
+            TELEMETRY_TYPE_NAME,
+            __version__,
+    ) as api_client:
+        api_instance = AWSIntegrationApi(api_client)
+        try:
+            api_instance.create_aws_account(aws_account)
+        except ApiException as e:
+            LOG.error("Exception when calling AWSIntegrationApi->create_aws_account: %s\n", e)
+            return ProgressEvent(
+                status=OperationStatus.FAILED, resourceModel=model, message=f"Error creating AWS account: {e}"
+            )
+
+    integration_id = f"{model.AccountID}:{model.RoleName}:{model.AccessKeyID}"
+    model.IntegrationID = integration_id
+
+    return read_handler(session, request, callback_context)
+
+
+@resource.handler(Action.UPDATE)
+def update_handler(
+        session: Optional[SessionProxy],
+        request: ResourceHandlerRequest,
+        callback_context: MutableMapping[str, Any],
+) -> ProgressEvent:
+    LOG.info("Starting %s Update Handler", TYPE_NAME)
+    model = request.desiredResourceState
+
+    aws_account = build_aws_account_from_model(model)
+    with v1_client(
+            model.DatadogCredentials.ApiKey,
+            model.DatadogCredentials.ApplicationKey,
+            model.DatadogCredentials.ApiURL or "https://api.datadoghq.com",
+            TELEMETRY_TYPE_NAME,
+            __version__,
+    ) as api_client:
+        api_instance = AWSIntegrationApi(api_client)
+        try:
+            api_instance.update_aws_account(
+                aws_account,
+                account_id=model.AccountID,
```

Comment:
Does this update properly or would this fail if the account_id/role_name/access_key_id were to be updated? 
Do you think it would make sense to try and catch that based on the model.id and try to throw an error?